twitchAPI/oauth.py

```diff
@@ -372,11 +375,14 @@ async def authenticate(self,
             # wait for the server to start up
             while not self._server_running:
                 await asyncio.sleep(0.01)
-            # open in browser
-            browser = webbrowser.get(browser_name)
-            browser.open(self._build_auth_url(), new=browser_new)
-            while self._user_token is None:
-                await asyncio.sleep(0.01)
+            if use_browser:
+                # open in browser
+                browser = webbrowser.get(browser_name)
+                browser.open(self._build_auth_url(), new=browser_new)
+                while self._user_token is None:
+                    await asyncio.sleep(0.01)
+            else:
+                self.logger.info(f"To authenticate open: {self._build_auth_url()}")
```

Comment:
Not sure if logging the url is needed, because you can retrieve the url with `return_auth_url()` method from inside your own code.